drivers/sensor/lis2dh/lis2dh.c

```diff
@@ -98,6 +98,15 @@ static int lis2dh_channel_get(const struct device *dev,
 	int ofs_end;
 	int i;
 
+#if defined(CONFIG_LIS2DH_FIFO_MODE)
+	int ret;
+
+	ret = ring_buf_get(&lis2dh->fifo_rb, lis2dh->sample.xyz, 6);
+	if (ret != 6) {
+		return -ENODATA;
+	}
+#endif
```

Comment:
I have the same comment as in #43429. I think that FIFO should be enabled as per driver instance base.
LIS2DH driver is multi-instance, and we may have one instance configured in FIFO mode and the other not.

Reply:
Maybe we can ALWAYS use ring buffer even in non-FIFO case, which would fall in a ring buffer of size 1.
What do you think guys?

Reply:
Tbh in hindsight the ring buf seems a bit try-hardy. It relieves the application of responsibility of completely flushing the buffer with expediency, but what's the value in that really? I'd rather just require the app to be quick with channel_get.

Another option: introduce a device specific channel for "32 x xyz samples". Could be argued this is the least dirty in terms of API abuse.

Reply:
Actually, my current implementation breaks the promises set up by `sensor.h`. `channel_get` if called twice in a row SHALL return the same sample unless `sample_fetch` was called in between.

In other words, this needs work. Still needs to be decided if a custom channel is better than forcing the application to loop-call `channel_get + sample_fetch` thirty-two times (or alternatively, if the driver should loop-call a DRDY handler thirty-two times)

This is messy business :P